src/app/shared/components/filters/filters.component.ts

```diff
@@ -50,18 +60,36 @@ export class FiltersComponent {
     return FilterType;
   }
 
-  public get isFiltered() {
+  public get areFiltersChangedSinceLastApply(): boolean {
+    return this._areFiltersChangedSinceLastApply;
+  }
+
+  public get isFilterNotEmpty() {
     return !!this._filters.find(f => f.value);
   }
 
+  public ngOnChanges(changes: SimpleChanges): void {
+    if (changes['filters']) {
+      this._appliedFilters = _.cloneDeep(this._filters);
+      this.checkFiltersChange();
+    }
+  }
+
   public apply() {
+    this._appliedFilters = _.cloneDeep(this._filters);
+    this.checkFiltersChange();
```

Comment:
#nit: Line-of-Codes:79-80 are duplicated in two places, may be better to separate them to method?